### ISSUE TITLE
Data Dictionary Export fails for projects with over 2000 fields.

### ISSUE THREAD

**Issue by burrowse** [OuhscBbmc/REDCapR#221]
Hello,

First of all, thank you so much for creating the REDCapR package!!

We ran into an issue where we get the following message:

```
The data dictionary describing 0 fields was read from REDCap in 2.4 seconds.  The http status code was 200.
```
We are, however, able to export data from these projects just fine. We were hoping to use the data dictionary to filter and retrieve all fields associated with various forms as 'form' does not seem to be an option on redcap_read.

Has anyone run into this issue before? Any assistance is appreciated!

**Comment by wibeasley** [comment]
@thomasnwilson reported that problem a few months ago.  We started investigating it w/ https://github.com/OuhscBbmc/REDCapR/tree/master/inst/test-data/super-wide.  

Can you upload that project and tell me what you guys get?  What version of REDCap?

**Comment by burrowse** [comment]
Thanks for getting back to me so quickly!

We are on version 8.3.2.

I uploaded that project and I am able to successfully export from that project:

```
 ds <- redcap_metadata_read(redcap_uri=redCapURI, token=readToken)
The data dictionary describing 3,001 fields was read from REDCap in 1 seconds.  The http status code was 200
```

One thing I did notice that in your test project, all of the fields are on one form and this is not the case in the projects that I have tried this out in.

**Comment by wibeasley** [comment]
The form count could be important.  It's tough to know what corner cases could make it fail.  I've added two more forms (in the [dev branch](https://github.com/OuhscBbmc/REDCapR/tree/dev/inst/test-data/super-wide); one for each 1k columns) but it's still working with manual calling and a handful of [new unit tests](https://github.com/OuhscBbmc/REDCapR/commit/061d338f1dfffbdc90ab1ceb36ef851a6f96ab49) on our REDCap 8.4.0 instance.

```r
> redcap_metadata_read(redcap_uri=credential_super_wide$redcap_uri, token=credential_super_wide$token)
The data dictionary describing 3,001 fields was read from REDCap in 0.7 seconds.  The http status code was 200.
```

But I believe you that it's failing with your project in your environment. Sorry this super-wide example doesn't produce the same error.  If you add/modify that data dictionary and phi-free data file, I'll see if it reproduces on our server, and I'll modify REDCapR if necessary.

----

> We were hoping to use the data dictionary to filter and retrieve all fields associated with various forms as 'form' does not seem to be an option on redcap_read.

Big picture though, make sure you're up-to-date on R and all the packages (especially httr and readr).  It appears that you're using an old version of REDCapR, because we added the 'form' filter a few weeks ago (see #206).  It hasn't made it to CRAN yet.  I was making sure the master branch of GitHub was running smoothly for a few weeks before I resubmit it to CRAN.

https://github.com/OuhscBbmc/REDCapR#download-and-installation-instructions
